kornia/color/hsv.py

```diff
@@ -27,7 +27,16 @@ def rgb_to_hsv(image: torch.Tensor) -> torch.Tensor:
         raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
                          .format(image.shape))
 
-    maxc, max_indices = image.max(-3)
+    # TODO: enable again for later versions than 1.6.0 or find a different implementation.
+    # It turns out that .max(...) does not return the index in the first position when
+    # all the inputs have the same value in CUDA.
+    # maxc, max_indices = image.max(-3)
+    if image.is_cuda and torch.__version__ == '1.6.0':
+        maxc, max_indices = image.cpu().max(-3)
+        maxc, max_indices = maxc.to(image), max_indices.to(image.device)
```

Comment:
@shijianjian @pmeier @edgarriba @FGeri had to do this hack since somehow cuda .max have weird behaviour

Reply:
Why is this a requirement? Shouldn't this be irrelevant at which index the max value is?

Reply:
this was an optimization done by @FGeri 
https://github.com/kornia/kornia/commit/56acb451088f67b0f00212f2e7d74df8eea189ee#diff-574268a035fea7cbc4fc3554acd1f0c40c6e8d42d2b7918298ef209f91f71dc9